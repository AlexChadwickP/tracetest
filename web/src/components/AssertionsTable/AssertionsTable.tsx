import {Table} from 'antd';
import {ColumnsType} from 'antd/lib/table';
import {AssertionResult} from 'types';

interface IProps {
  assertionResults: AssertionResult[];
}

const AssertionsResultTable = ({assertionResults}: IProps) => {
  const data = assertionResults.map(el => {
    return {
      key: el.spanAssertionId,
      name: el.selector,
      selectedSpans: el.spanCount,
      property: el.propertyName,
      comparison: el.operator,
      value: el.comparisonValue,
      results: `${el.passedSpanCount}/${el.spanCount - el.passedSpanCount}`,
    };
  });

  const columns: ColumnsType<any> = [
    {
      title: 'Span selector',
      dataIndex: 'name',
      key: 'name',
      render: (value, record, index) => {
        const obj = {
          children: value,
          props: {rowSpan: 1},
        };
        if (data.filter(el => el.name === value).length === 1) {
          return obj;
        }
        if (data.findIndex(el => el.name === value) === index) {
          const count = data.filter(item => item.name === value).length;
          obj.props.rowSpan = count;
          return obj;
        }
        obj.props.rowSpan = 0;
        return obj;
      },
    },
    {
      title: '# Selected',
      dataIndex: 'selectedSpans',
      key: 'selectedSpans',
    },

    {
      title: 'Property',
      dataIndex: 'property',
      key: 'property',
    },
    {
      title: 'Comparison',
      dataIndex: 'comparison',
      key: 'comparison',
    },
    {
      title: 'Value',
      dataIndex: 'value',
      key: 'value',
    },
    {
      title: 'Pass/Fail',
      dataIndex: 'results',
      key: 'results',
    },
  ];

  return <Table size="small" pagination={{hideOnSinglePage: true}} dataSource={data} columns={columns} />;
};

export default AssertionsResultTable;
