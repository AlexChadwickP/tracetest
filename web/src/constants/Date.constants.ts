export const RESULT_DATE_FORMAT = "EEEE, do MMMM yyyy 'at' HH:mm:ss";
