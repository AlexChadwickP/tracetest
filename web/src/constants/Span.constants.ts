export const HttpRequestAttributeList = [
  'http.method',
  'http.url',
  'http.target',
  'http.host',
  'http.scheme',
  'http.request_content_length',
  'http.request_content_length_uncompressed',
  'http.retry_count exists',
  'http.user_agent',
];

export const HttpResponseAttributeList = ['http.status_code'];
