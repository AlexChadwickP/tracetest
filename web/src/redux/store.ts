import {configureStore, ThunkAction, Action} from '@reduxjs/toolkit';
import {testAPI} from '../services/TestService';

export const store = configureStore({
  reducer: {
    [testAPI.reducerPath]: testAPI.reducer,
  },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
