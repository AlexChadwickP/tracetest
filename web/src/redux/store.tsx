import {Action, configureStore, ThunkAction} from '@reduxjs/toolkit';
import TestAPI from 'redux/apis/TraceTest.api';

export const store = configureStore({
  reducer: {
    [TestAPI.reducerPath]: TestAPI.reducer,
  },
  middleware: getDefaultMiddleware => getDefaultMiddleware().concat(TestAPI.middleware),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<ReturnType, RootState, unknown, Action<string>>;
