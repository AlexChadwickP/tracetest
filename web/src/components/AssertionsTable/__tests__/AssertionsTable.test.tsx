import {render} from '@testing-library/react';
import {SemanticGroupNames} from 'constants/SemanticGroupNames.constants';
import {Provider} from 'react-redux';
import {CompareOperator} from '../../../constants/Operator.constants';
import {LOCATION_NAME} from '../../../constants/Span.constants';
import {SpanAttributeType} from '../../../constants/SpanAttribute.constants';
import {store} from '../../../redux/store';
import AssertionsTable from '../index';

test('AssetionsTable', () => {
  const result = render(
    <Provider store={store}>
      <AssertionsTable
        assertionResults={[
          {
            actualValue: '',
            comparisonValue: '',
            hasPassed: false,
            locationName: LOCATION_NAME.INSTRUMENTATION_LIBRARY,
            operator: CompareOperator.EQUALS,
            propertyName: '',
            spanAssertionId: '',
            spanId: '',
            valueType: SpanAttributeType.doubleValue,
          },
        ]}
        sort={1}
        assertion={{
          assertionId: '',
          selectors: [
            {
              locationName: LOCATION_NAME.INSTRUMENTATION_LIBRARY,
              propertyName: '',
              value: '',
              valueType: '',
            },
          ],
          spanAssertions: [],
        }}
        span={{
          attributeList: [],
          attributes: {},
          duration: 0,
          endTimeUnixNano: '',
          instrumentationLibrary: {name: '', version: ''},
          kind: '',
          name: '',
          parentSpanId: '',
          signature: [],
          spanId: '',
          startTimeUnixNano: '',
          status: {code: ''},
          traceId: '',
          type: SemanticGroupNames.Http,
        }}
        resultId="546"
        testId="234"
      />
    </Provider>
  );
  expect(result.container).toMatchSnapshot();
});
