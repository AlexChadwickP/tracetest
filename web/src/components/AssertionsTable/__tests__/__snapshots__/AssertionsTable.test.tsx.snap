// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AssetionsTable 1`] = `
<div>
  <div
    class="sc-hKwDye iytRYn"
    data-cy="assertion-table"
  >
    <div
      class="sc-eCImPb gTzCre"
    >
      <h5
        class="ant-typography"
        style="margin: 0px;"
      >
        Assertion #
        1
        <span
          class="ant-badge ant-badge-not-a-wrapper sc-jRQBWg kvNTIw"
        />
      </h5>
      <button
        class="ant-btn ant-btn-link"
        data-cy="edit-assertion-button"
        type="button"
      >
        <span>
          Edit
        </span>
      </button>
    </div>
    <div
      class="ant-table-wrapper sc-dkPtRN exmISC"
    >
      <div
        class="ant-spin-nested-loading"
      >
        <div
          class="ant-spin-container"
        >
          <div
            class="ant-table ant-table-small ant-table-bordered ant-table-layout-fixed"
          >
            <div
              class="ant-table-container"
            >
              <div
                class="ant-table-content"
              >
                <table
                  style="table-layout: fixed;"
                >
                  <colgroup>
                    <col
                      style="width: 50%;"
                    />
                  </colgroup>
                  <thead
                    class="ant-table-thead"
                  >
                    <tr>
                      <th
                        class="ant-table-cell ant-table-cell-ellipsis"
                        title="Property"
                      >
                        Property
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Comparison
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Value
                      </th>
                      <th
                        class="ant-table-cell"
                      >
                        Actual
                      </th>
                    </tr>
                  </thead>
                  <tbody
                    class="ant-table-tbody"
                  >
                    <tr
                      class="ant-table-row ant-table-row-level-0"
                      data-row-key=""
                    >
                      <td
                        class="ant-table-cell ant-table-cell-ellipsis"
                        title=""
                      >
                        <span
                          data-cy="assertion-check-property"
                        />
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        eq
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        
                      </td>
                      <td
                        class="ant-table-cell"
                      >
                        <span
                          class="ant-typography ant-typography-danger"
                        >
                          <strong>
                            New request
                          </strong>
                        </span>
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
